### ISSUE TITLE
IRestRequest.AddFile using a path requires too broad file permissions

### ISSUE THREAD

**Issue by g2petter** [restsharp/RestSharp#1272]
## Expected Behavior
Using the method with the signature `public IRestRequest AddFile(string name, string path, string contentType = null)` it should be possible to attach a file to a request when you only have read access on said file.

## Actual Behavior
Executing the request will fail once the `FileParameter.Writer` gets executed.

This happens because of the following code block in the method: 

```
                    using (var file = new StreamReader(new FileStream(path, FileMode.Open)))
                    {
                        file.BaseStream.CopyTo(s);
                    }
```

`FileMode.Open` assumes both read _and_ write permissions.

To fix this, the `FileStream` constructor should be called like this instead: `new FileStream(path, FileMode.Open, FileAccess.Read)`

## Steps to Reproduce the Problem

  1. Create a file on disk that your IIS user only has read access to
  2. Attach that file to an IRestRequest with the AddFile method that takes the file's path as a parameter
  3. Attempt to execute the request

## Specifications

  - Version: Tested with both 106.6.7 (most recent) and a couple of earlier versions
  - Platform: Windows
  - Subsystem: Windows 10, Windows Server 2016

**Comment by parag-patil** [comment]
Why this is closed without suggestions? I am also facing same issue.

**Comment by alexeyzimarev** [comment]
@parag-patil it is closed because the PR that fixes the issue was merged, as you can see per referenced commit.

**Comment by alexeyzimarev** [comment]
It is released in 106.6.10. Sorry about the delay.
